src/controllers/mainController.ts

```diff
@@ -185,22 +185,30 @@ export default class MainController implements vscode.Disposable {
                 this._scriptingService = new ScriptingService(this._connectionMgr, this._vscodeWrapper);
                 this._context.subscriptions.push(
                     vscode.commands.registerCommand(
-                        Constants.cmdScriptSelect, async (node: TreeNodeInfo) => {
-                    const uri = await this._untitledSqlDocumentService.newQuery();
-                    if (!this.connectionManager.isConnected(uri.toString())) {
-                        let connectionCreds = node.connectionCredentials;
-                        const databaseName = self.getDatabaseName(node);
-                        connectionCreds.database = databaseName;
-                        this._statusview.languageFlavorChanged(uri.toString(), Constants.mssqlProviderName);
-                        await this.connectionManager.connect(uri.toString(), connectionCreds);
-                        this._statusview.sqlCmdModeChanged(uri.toString(), false);
-                        const selectStatement = await this._scriptingService.scriptSelect(node, uri.toString());
+                        Constants.cmdScriptSelect, (node: TreeNodeInfo) => {
+                    return this._untitledSqlDocumentService.newQuery().then((uri) => {
                         let editor = this._vscodeWrapper.activeTextEditor;
-                        editor.edit(editBuilder => {
-                            editBuilder.replace(editor.selection, selectStatement);
-                        }).then(() => this.onRunQuery());
-                        return this.connectionManager.connectionStore.removeRecentlyUsed(<IConnectionProfile>connectionCreds);
-                    }
+                        if (editor) {
+                            let connectionCreds = node.connectionCredentials;
+                            const databaseName = self.getDatabaseName(node);
+                            connectionCreds.database = databaseName;
+                            this._statusview.languageFlavorChanged(uri.toString(), Constants.mssqlProviderName);
+                            return this.connectionManager.connect(uri.toString(), connectionCreds).then(() => {
```

Comment:
I would recommend rewriting this code to use `await`.  I think it would be significantly cleaner.